src/IO/PeekableReadBuffer.cpp

```diff
@@ -113,12 +113,21 @@ void PeekableReadBuffer::rollbackToCheckpoint(bool drop)
 {
     checkStateCorrect();
 
-    if (!checkpoint)
-        throw DB::Exception("There is no checkpoint", ErrorCodes::LOGICAL_ERROR);
```

Comment:
Ok. We can just remove [the check](https://github.com/ClickHouse/ClickHouse/blob/27600064159ac63e05800ef37afa41b317771352/src/IO/tests/gtest_peekable_read_buffer.cpp#L51-L64) for this.